upup/pkg/fi/cloudup/awstasks/network_load_balancer.go

```diff
@@ -571,10 +581,18 @@ func (_ *NetworkLoadBalancer) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *Ne
 		request.Type = e.Type
 
 		for _, subnetMapping := range e.SubnetMappings {
-			request.SubnetMappings = append(request.SubnetMappings, &elbv2.SubnetMapping{
-				SubnetId:           subnetMapping.Subnet.ID,
-				PrivateIPv4Address: subnetMapping.PrivateIPv4Address,
-			})
+      if subnetMapping.PrivateIPv4Address != nil {
```

Comment:
this needs to use tabs rather than spaces, and could be simplified to:

```suggestion
			request.SubnetMappings = append(request.SubnetMappings, &elbv2.SubnetMapping{
				SubnetId:           subnetMapping.Subnet.ID,
				PrivateIPv4Address: subnetMapping.PrivateIPv4Address,
				AllocationId:       subnetMapping.AllocationID,
			})
```

given that both the private ip and allocation id are pointers, assigning them nil values is equivalent to not assigning them at all.